distributed/executor.py

```diff
@@ -974,6 +985,46 @@ def cancel(self, futures):
         """
         return sync(self.loop, self._cancel, futures)
 
+    def publish_dataset(self, data, name):
+        """
+        Store a named reference to the data in the scheduler for other executors
+
+        Parameters
+        ----------
+        data: list of futures or dask collection
+        name: str or container
+            handle by which the stored data should be known. If not a string,
+            will be wrapped by tokey() to deterministically turn into one.
+
+        Returns
+        -------
+        None
+        """
+        keys = [tokey(f.key) for f in futures_of(data)]
+        self._send_to_scheduler({'op': 'publish-data', 'keys': keys,
+                                 'data': pickle.dumps(data),
+                                 'name': tokey(name)})
+
+    @gen.coroutine
+    def _published_datasets(self):
+        resp = yield self.scheduler.get_published_keys()
```

Comment:
Perhaps this should be named `published_datasets` on the scheduler as well to align terms?  

We might also consider getting rid of this method and just using `sync(self.loop, self.scheduler.published_datasets)` directly.